Fix NestedFrom not support validatorTrigger

yhsun315 opened the pull request:
add validatorTrigger props to NestedFrom level, the validator works nicely when I play around with the ComplexFrom, also some small fix around the CurrecyField type

valendres commented on services/demo/src/view/forms/ComplexForm.tsx:
```diff
@@ -13,17 +13,23 @@ import {
-const lessThanCurrencyField = (targetFieldName: keyof ComplexFormFields) => (
+const lessThanCurrencyField = (
+    targetFieldName: keyof ComplexFormFields,
+    value: string,
```
Why do we need to pass in the value here?

yhsun315 replied:
the values here are the NestedFrom values, not the parent From values, so it could not locate the values[targetFieldName]

valendres commented on packages/core/src/components/NestedForm/NestedForm.tsx:
```diff
@@ -10,6 +10,7 @@ import { BoundComponent } from '../bind';
 import { OutsideFormError } from '../../errors';
 
 export interface NestedFormProps {
+    validatorTrigger?: string;
```
How is this prop being used?

yhsun315 replied:
Through debugging, this is helping  down the way in `Form.tsx=>getComponentValidatorTriggers=>getComponentProps(componentName)` to get the `validatorTrigger` props, without this `props.validatorTrigger` will always undefined

valendres reviewed (commented):
Would you be able to write a unit test for this to assert that the desired behaviour is working correctly?

yhsun315 commented:
> Would you be able to write a unit test for this to assert that the desired behaviour is working correctly?

I added a unit test but marked as skip sadly, there seems to be a problem with the Mirror is not getting the values, I've added some comment in the test, appreciate if you can help debug a bit.
